bindings/rust/src/joypad_nintendo.rs

```diff
@@ -0,0 +1,71 @@
+use std::ffi::{c_int, c_void};
+use std::path::PathBuf;
+use crate::ffi::{
+    inputtino_joypad_switch_create,
+    inputtino_joypad_switch_destroy,
+    inputtino_joypad_switch_get_nodes,
+    inputtino_joypad_switch_set_on_rumble,
+    inputtino_joypad_switch_set_pressed_buttons,
+    inputtino_joypad_switch_set_stick,
+    inputtino_joypad_switch_set_triggers,
+};
+use crate::common::{get_nodes, make_device, DeviceDefinition};
+use crate::JoypadStickPosition;
+
+pub struct SwitchJoypad {
+    joypad: *mut crate::ffi::InputtinoSwitchJoypad,
+    on_rumble_fn: Box<dyn FnMut(i32, i32) -> ()>,
+}
+
+impl SwitchJoypad {
+    pub fn new(device: &DeviceDefinition) -> Result<Self, String> {
+        make_device(inputtino_joypad_switch_create, device)
+            .map(|joypad| SwitchJoypad { joypad, on_rumble_fn: Box::new(|_, _| {}) })
+    }
+
+    pub fn get_nodes(&self) -> Result<Vec<PathBuf>, String> {
+        get_nodes(inputtino_joypad_switch_get_nodes, self.joypad)
+    }
+
+    pub fn set_pressed(&self, buttons: i32) {
```

Comment:
The fact that each of these types has the exact same methods, but that is not enforced by the compiler, indicates that they should just be one type, imo - either a trait with many impls or just one `Joypad` type.

Reply:
I think a trait with many impls is closer to the C API. Without changing the C/C++ API too, I think that makes more sense.